In [7]:
import pandas as pd
import geopandas as gpd

In [3]:
park_crime=pd.read_csv('../data/1_parkcrime.csv')

In [10]:
shape_path="../data/2 park_location.geojson"
park_shape = gpd.GeoDataFrame.from_file(shape_path).to_crs(epsg=4326)

In [16]:
park_shape.park_name= park_shape.park_name.str.lower()

In [21]:
location = park_shape[['park_name','geometry']]
location = location.rename(columns={'park_name':'PARK'})

In [34]:
score = pd.read_csv('../data/3_park_score.csv')
score = score[['SiteName','average_score']]
score = score.rename(columns={"SiteName":'PARK'})

In [40]:
score.PARK=score.PARK.str.lower()

In [67]:
len(park_crime.PARK.value_counts())

743

#### We have crime record for 743 parks in total.

In [49]:
len(park_crime.merge(score,on='PARK',how='inner')['PARK'].value_counts())

492

#### If we use park score, inner merge, only 492 parks left.

In [42]:
park_crime.merge(score,on='PARK',how='outer')

PERIOD                                  PARK        BOROUGH  \
0     2014-q3                       pelham bay park  BRONX           
1     2014-q4                       pelham bay park  BRONX           
2     2015-q1                       pelham bay park  BRONX           
3     2015-q2                       pelham bay park  BRONX           
4     2015-q3                       pelham bay park  BRONX           
5     2015-q4                       pelham bay park  BRONX           
6     2016-q1                       pelham bay park  BRONX           
7     2016-q2                       pelham bay park  BRONX           
8     2016-q3                       pelham bay park  BRONX           
9     2014-q3                    van cortlandt park  BRONX           
10    2014-q4                    van cortlandt park  BRONX           
11    2015-q1                    van cortlandt park  BRONX           
12    2015-q2                    van cortlandt park  BRONX           
13    2015-q3                    van cortlandt park  BRONX           
14    2015-q4                    van cortlandt park  BRONX           
15    2016-q1                    van cortlandt park  BRONX           
16    2016-q2                    van cortlandt park  BRONX           
17    2016-q3                    van cortlandt park  BRONX           
18    2014-q3          rockaway beach and boardwalk  QUEENS          
19    2014-q4          rockaway beach and boardwalk  QUEENS          
20    2015-q1          rockaway beach and boardwalk  QUEENS          
21    2015-q2          rockaway beach and boardwalk  QUEENS          
22    2015-q3          rockaway beach and boardwalk  QUEENS          
23    2015-q4          rockaway beach and boardwalk  QUEENS          
24    2016-q1          rockaway beach and boardwalk  QUEENS          
25    2016-q2          rockaway beach and boardwalk  QUEENS          
26    2016-q3          rockaway beach and boardwalk  QUEENS          
27    2014-q3                       freshkills park  STATEN ISLAND   
28    2014-q4                       freshkills park  STATEN ISLAND   
29    2015-q1                       freshkills park  STATEN ISLAND   
...       ...                                   ...            ...   
7929      NaN                   saw mill playground            NaN   
7930      NaN                         youth village            NaN   
7931      NaN                    estella diggs park            NaN   
7932      NaN                      grand playground            NaN   
7933      NaN                         barretto park            NaN   
7934      NaN  kingsbridge heights community center            NaN   
7935      NaN                 mount hope playground            NaN   
7936      NaN                  cleopatra playground            NaN   
7937      NaN                  field of dreams park            NaN   
7938      NaN                    galileo playground            NaN   
7939      NaN                   eae j mitchell park            NaN   
7940      NaN                      cpl fischer park            NaN   
7941      NaN                          ranaqua park            NaN   
7942      NaN                      rock garden park            NaN   
7943      NaN                     mount hope garden            NaN   
7944      NaN                         boston garden            NaN   
7945      NaN                  stop & go playground            NaN   
7946      NaN                          seabury park            NaN   
7947      NaN                  beanstalk playground            NaN   
7948      NaN             rev lena irons unity park            NaN   
7949      NaN             ogden plimpton playground            NaN   
7950      NaN                     west farms rapids            NaN   
7951      NaN                           walton park            NaN   
7952      NaN                         morris garden            NaN   
7953      NaN                half-nelson playground            NaN   
7954      NaN    

In [58]:
demo = pd.read_csv('../data/4 cleaned_medianincome_houldhold.csv')
demo = demo[['park_name','household','median_income']]
demo= demo.rename(columns={'park_name':'PARK'})
demo.PARK = demo.PARK.str.lower()

In [65]:
park_crime = park_crime.merge(demo,on='PARK',how = 'inner')

In [73]:
complaints = pd.read_csv('../data/6 311_complains_merged_with_crime.csv')
del complaints['Unnamed: 0']

In [78]:
crime_311_income = complaints.merge(demo,on='PARK',how='outer')

In [81]:
crime_311_income_score = crime_311_income.merge(score,on='PARK')
crime_311_income_score.head()

PERIOD        PARK        BOROUGH  SIZE (ACRES)            CATEGORY  \
0  2014-q3  bronx park  BRONX               718.373  ONE ACRE OR LARGER   
1  2014-q4  bronx park  BRONX               718.373  ONE ACRE OR LARGER   
2  2015-q1  bronx park  BRONX               718.373  ONE ACRE OR LARGER   
3  2015-q2  bronx park  BRONX               718.373  ONE ACRE OR LARGER   
4  2015-q3  bronx park  BRONX               718.373  ONE ACRE OR LARGER   

   MURDER  RAPE  ROBBERY  FELONY ASSAULT  BURGLARY  GRAND LARCENY  \
0     0.0   0.0      1.0             0.0       0.0            2.0   
1     0.0   1.0      0.0             0.0       0.0            0.0   
2     0.0   0.0      0.0             0.0       0.0            0.0   
3     0.0   0.0      0.0             1.0       0.0            3.0   
4     1.0   0.0      1.0             2.0       0.0            2.0   

   GRAND LARCENY OF MOTOR VEHICLE  CRIME_TOT  Street_light  Drinking  Drugs  \
0                             0.0        3.0          17.0       1.0    0.0   
1                             0.0        1.0          20.0       0.0    0.0   
2                             0.0        0.0          10.0       0.0    0.0   
3                             0.0        4.0          24.0       0.0    0.0   
4                             0.0        6.0          16.0       1.0    0.0   

   311_Total  household  median_income  average_score  
0       18.0     1069.0        34453.0          0.625  
1       20.0     1069.0        34453.0          0.625  
2       10.0     1069.0        34453.0          0.625  
3       24.0     1069.0        34453.0          0.625  
4       17.0     1069.0        34453.0          0.625

In [86]:
public_h = pd.read_csv('../data/7 cleaned nyc public house.csv')
del public_h['Unnamed: 0']

In [91]:
public_h= public_h.rename(columns={'park_name':"PARK"})

In [96]:
public_h.PARK = public_h.PARK.str.lower()

In [156]:
crime_311_income_score_pubh =crime_311_income_score.merge(public_h,on='PARK')

In [157]:
crime_311_income_score_pubh = crime_311_income_score_pubh[crime_311_income_score_pubh.CATEGORY=='ONE ACRE OR LARGER']

In [158]:
crime_311_income_score_pubh = crime_311_income_score_pubh.fillna(0)

In [159]:
crime_311_income_score_pubh = crime_311_income_score_pubh.drop_duplicates() 

In [161]:
crime_311_income_score_pubh= crime_311_income_score_pubh.reset_index(drop= True)

In [163]:
crime_311_income_score_pubh

PERIOD                 PARK        BOROUGH  SIZE (ACRES)  \
0     2014-q3           bronx park  BRONX               718.373   
1     2014-q4           bronx park  BRONX               718.373   
2     2015-q1           bronx park  BRONX               718.373   
3     2015-q2           bronx park  BRONX               718.373   
4     2015-q3           bronx park  BRONX               718.373   
5     2015-q4           bronx park  BRONX               718.373   
6     2016-q1           bronx park  BRONX               718.373   
7     2016-q2           bronx park  BRONX               718.370   
8     2016-q3           bronx park  BRONX               718.370   
9     2014-q3     silver lake park  STATEN ISLAND       205.977   
10    2014-q4     silver lake park  STATEN ISLAND       205.977   
11    2015-q1     silver lake park  STATEN ISLAND       205.977   
12    2015-q2     silver lake park  STATEN ISLAND       205.977   
13    2015-q3     silver lake park  STATEN ISLAND       205.977   
14    2015-q4     silver lake park  STATEN ISLAND       205.977   
15    2016-q1     silver lake park  STATEN ISLAND       205.977   
16    2016-q2     silver lake park  STATEN ISLAND       205.980   
17    2016-q3     silver lake park  STATEN ISLAND       205.980   
18    2014-q3    blood root valley  STATEN ISLAND       161.242   
19    2014-q4    blood root valley  STATEN ISLAND       161.242   
20    2015-q1    blood root valley  STATEN ISLAND       161.242   
21    2015-q2    blood root valley  STATEN ISLAND       161.242   
22    2015-q3    blood root valley  STATEN ISLAND       161.242   
23    2015-q4    blood root valley  STATEN ISLAND       161.242   
24    2016-q1    blood root valley  STATEN ISLAND       161.242   
25    2016-q2    blood root valley  STATEN ISLAND       161.240   
26    2016-q3    blood root valley  STATEN ISLAND       161.240   
27    2014-q3       high rock park  STATEN ISLAND        94.100   
28    2014-q4       high rock park  STATEN ISLAND        94.100   
29    2015-q1       high rock park  STATEN ISLAND        94.100   
...       ...                  ...            ...           ...   
4254  2016-q1      pink playground  BROOKLYN              1.000   
4255  2016-q2      pink playground  BROOKLYN              1.000   
4256  2016-q3      pink playground  BROOKLYN              1.000   
4257  2014-q3       playground 174  BRONX                 1.000   
4258  2014-q4       playground 174  BRONX                 1.000   
4259  2015-q1       playground 174  BRONX                 1.000   
4260  2015-q2       playground 174  BRONX                 1.000   
4261  2015-q3       playground 174  BRONX                 1.000   
4262  2015-q4       playground 174  BRONX                 1.000   
4263  2016-q1       playground 174  BRONX                 1.000   
4264  2016-q2       playground 174  BRONX                 1.000   
4265  2016-q3       playground 174  BRONX                 1.000   
4266  2014-q3  plimpton playground  BRONX                 1.000   
4267  2014-q4  plimpton playground  BRONX                 1.000   
4268  2015-q1  plimpton playground  BRONX                 1.000   
4269  2015-q2  plimpton playground  BRONX                 1.000   
4270  2015-q3  plimpton playground  BRONX                 1.000   
4271  2015-q4  plimpton playground  BRONX                 1.000   
4272  2016-q1  plimpton playground  BRONX                 1.000   
4273  2016-q2  plimpton playground  BRONX                 1.000   
4274  2016-q3  plimpton playground  BRONX                 1.000   
4275  2014-q3   randall playground  BRONX                 1.000   
4276  2014-q4   randall playground  BRONX                 1.000   
4277  2015-q1   randall playground  BRONX                 1.000   
4278  2015-q2   randall playground  BRONX                 1.000   
4279  2015-q3   randall playground  BRONX                 1.000   
4280  2015-q4   randall playground  BRONX                 1.000   
4281  2016-q1   randall playground  BRONX             

In [164]:
business=pd.read_csv('../data/5-park_business.csv')

In [166]:
del business['Unnamed: 0']

In [173]:
business = business.drop_duplicates()
business= business.rename(columns= {"park_name":"PARK"})
business.PARK = business.PARK.str.lower()

In [175]:
business= business.rename(columns= {"bus_act_count":"business"})

In [179]:
len(crime_311_income_score_pubh)

4284

In [180]:
crime_311_income_score_pubh_busi = crime_311_income_score_pubh.merge(business,on='PARK',how= 'outer')

In [183]:
crime_311_income_score_pubh_busi= crime_311_income_score_pubh_busi[crime_311_income_score_pubh_busi.CATEGORY=='ONE ACRE OR LARGER']

In [185]:
crime_311_income_score_pubh_busi.to_csv('../data/0_merged_data.csv')